mono/eglib/glib.h

```diff
@@ -1148,4 +1408,24 @@ glong     g_utf8_pointer_to_offset (const gchar *str, const gchar *pos);
  
 G_END_DECLS
 
-#endif
+// For each allocator; i.e. returning gpointer that needs to be cast.
+// Macros do not recurse, so naming function and macro the same is ok.
+// However these are also already macros.
+#undef g_malloc
+#undef g_realloc
+#undef g_malloc0
+#undef g_calloc
+#undef g_try_malloc
+#undef g_try_realloc
+#undef g_memdup
+#undef g_hash_table_lookup
+#define g_malloc(x) (g_cast (monoeg_malloc (x)))
+#define g_realloc(obj, size) (g_cast (monoeg_realloc ((obj), (size))))
+#define g_malloc0(x) (g_cast (monoeg_malloc0 (x)))
+#define g_calloc(x, y) (g_cast (monoeg_g_calloc ((x), (y))))
+#define g_try_malloc(x) (g_cast (monoeg_try_malloc (x)))
+#define g_try_realloc(obj, size) (g_cast (monoeg_try_realloc ((obj), (size))))
+#define g_memdup(mem, size) (g_cast (monoeg_g_memdup ((mem), (size))))
+#define g_hash_table_lookup(hash, key) (g_cast (monoeg_g_hash_table_lookup ((hash), (key))))
```

Comment:
See for example

https://github.com/mono/mono/pull/10019/files#diff-98cf043c47e9ac369f935d88ea2c0e5dR517

https://github.com/mono/mono/pull/10019/files#diff-34ddcfc99bebbe2859c2bafcd20a2ac4R1320

for the slightly larger use.